client/src/legacy/HtmlEditorField.js

```diff
@@ -187,6 +187,15 @@ ss.editorWrappers.tinyMCE = (function() {
       return this.getInstance().getContent();
     },
 
+    getSelection: function() {
+      const instance = this.getInstance();
+      let selection = instance.selection.getSel().toString();
+      if (!selection && instance.selection.getSel().type === 'Range') {
+        selection = instance.selection.getNode().outerHTML;
+      }
```

Comment:
This doesn't seem to be correct. In CMS 4 this still results in the buggy behaviour. When I debug this with my image selected, `instance.selection.getSel()` returns:

```
Selection {
  anchorNode: img.leftAlone.ss-htmleditorfield-file.image,
  anchorOffset: 0,
  focusNode: img.leftAlone.ss-htmleditorfield-file.image,
  focusOffset: 0,
  isCollapsed: true,
  rangeCount: 1,
  type: "Caret",
  caretBidiLevel: 0
}
```

Calling `toString()` on this results in an empty string, and because the type isn't `Range` it returns that empty string.

If this works in CMS 5 it is likely due to the changes in tinymce - we will need to do a CMS 4 specific fix first, then merge up, then if it's still broken in CMS 5 we'll need to tackle that separately.

I've suggested a change below that seems to work fine, and doesn't require this new function at all.

Reply:
I still cannot reproduce this issue with CMS 4.13. And when I check value of `instance.selection.getSel()` on selected image I have the following:
```
Selection {
  anchorNode: p
  anchorOffset: 0
  baseNode: p
  baseOffset: 0
  extentNode: p
  extentOffset: 1
  focusNode: p
  focusOffset: 1
  isCollapsed: false
  rangeCount: 1
  ----> type: "Range"
}
```

Reply:
Looks like you're somehow selecting a paragraph element instead of the image... how are you selecting the image? For me I click on the image - it briefly selects and then deselects it, so I have to click it a second time to select it. Video to show that in action:
[Screencast from 04-09-23 15:51:15.webm](https://github.com/silverstripe/silverstripe-admin/assets/36352093/57331e16-753a-4d84-b9bd-118638eb4f4c)

Reply:
[This is record](https://github.com/silverstripe/silverstripe-admin/assets/87288324/8ef74cf2-dcf1-4d9b-9731-17c30a915134
) what happens in my local environment for clean install(silverstripe-installer) CMS 4.13